mindsdb/integrations/handlers/bard_handler/bard_handler.py

```diff
@@ -0,0 +1,103 @@
+from mindsdb.utilities import log
+from mindsdb.utilities.config import Config
+
+logger = log.getLogger(__name__)
+
+
+class BardHandler(BaseMLEngine):
+    """
+    Integration with the Anthropic LLM Python Library
+    """
+
+    name = "bard"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.default_chat_model = "gemini-pro"
+        self.supported_chat_models = ["gemini-pro"]
+        # self.default_max_tokens = 100
+        self.generative = True
+        self.connection = None
+
+    def create(
+        self,
+        target: str,
+        df: Optional[pd.DataFrame] = None,
+        args: Optional[Dict] = None,
+    ) -> None:
+        self.model_storage.json_set("args", args)
+
+    def predict(
+        self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None
+    ) -> None:
+        args = self.model_storage.json_get("args")
+        api_key = self._get_bard_api_key(args)
+        genai.configure(api_key=api_key)
+
+        input_column = args["using"]["column"]
+        if input_column not in df.columns:
+            raise RuntimeError(f'Column "{input_column}" not found in input data')
+        if "model" not in args["using"]:
+            args["using"]["model"] = self.default_chat_model
+
+        self.connection = genai.GenerativeModel(args["using"]["model"])
+        result_df = pd.DataFrame()
+        result_df["predictions"] = df[input_column].apply(self.predict_answer)
+        result_df = result_df.rename(columns={"predictions": args["target"]})
+        return result_df
+
+    def _get_bard_api_key(self, args, strict=True):
+        """
+        API_KEY preference order:
+            1. provided at model creation
+            2. provided at engine creation
+            3. GOOGLE_API_KEY env variable
+            4. bard.api_key setting in config.json
+        """
+
+        if "api_key" in args["using"]:
+            return args["using"]["api_key"]
+        # 2
+        connection_args = self.engine_storage.get_connection_args()
+        if "api_key" in connection_args:
+            return connection_args["api_key"]
+        # 3
+        api_key = os.getenv("GOOGLE_API_KEY")
+        if api_key is not None:
+            return api_key
+        # 4
+        config = Config()
+        bard_config = config.get("bard", {})
+        if "api_key" in bard_config:
+            return bard_config["api_key"]
+
+        if strict:
+            raise Exception(
+                'Missing API key "api_key". Either re-create this ML_ENGINE specifying the `api_key` parameter,\
+                 or re-create this model and pass the API key with `USING` syntax.'
+            )
+
+    def predict_answer(self, text):
+        """
+        connects with bard api to predict the answer for the particular question
+
+        """
+
+        args = self.model_storage.json_get("args")
+        if "ai_prompt" not in args["using"]:
+            args["using"]["ai_prompt"] = ""
```

Comment:
Could you explain the reasoning behind the following prompt template?

```f"{HUMAN_PROMPT} {text} {AI_PROMPT}"```

I'm not sure I follow.

Assuming it is what we want, additionallyy are these empty prompts good default values? Seems like it could lead to poor model completions.

Reply:
I referred the anthropic handler while creating this one that's why I included this